jsonnet/lib/thanos-receive-controller.libsonnet

```diff
@@ -1,125 +1,165 @@
 local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
 
 {
-  _config+:: {
-    namespace:: 'observatorium',
-    version:: 'latest',
-    imageRepo:: 'quay.io/observatorium/thanos-receive-controller',
+  local trc = self,
+
+  config:: {
+    name: error 'must provide name',
+    namespace: error 'must provide namespace',
+    version: error 'must provide version',
+    image: error 'must provide image',
+    replicas: error 'must provide replicas',
+    hashrings: error 'must provide hashring configuration',
+
+    commonLabels:: {
+      'app.kubernetes.io/name': 'thanos',
+      'app.kubernetes.io/instance': trc.config.name,
+      'app.kubernetes.io/version': 'v' + trc.config.version,
```

Comment:
This is probably only true for semver tags and only when those tags don't already start with a 'v'. I'd say let's not prepend anything to the version given by the user. "We expect you to provide the correct version" is easier to say than "We expect you to provide the correct version of your app unless that version starts with a 'v', in which case elide the first character, or if your version is not semver, in which case... x"

Reply:
fixed.